src/preferences/configobject.cpp

```diff
@@ -211,9 +215,21 @@ template <class ValueType> void ConfigObject<ValueType>::save() {
         }
         file.close();
         if (file.error()!=QFile::NoError) { //could be better... should actually say what the error was..
-            qDebug() << "Error while writing configuration file:" << file.errorString();
+            qWarning() << "Error while writing configuration file: " << file.errorString();
+        } else {
+            QFile oldConfig(m_filename);
+            if (!oldConfig.remove()) {
+                qWarning() << "Could not remove old config file: " << oldConfig.errorString();
+                return false;
+            }
+            if (file.rename(m_filename)) {
```

Comment:
Continue with the early return pattern from the preceding conditional here, i.e. `if (!file.rename(m_filename)) {`. Following different branching styles (success vs failure) is difficult otherwise.